tools/scylla-nodetool.cc

```diff
@@ -2644,6 +2651,257 @@ void table_stats_operation(scylla_rest_client& client, const bpo::variables_map&
+    auto res = client.get("/task_manager/list_modules");
+
+    for (auto& x: res.GetArray()) {
+        fmt::print("{}\n", rjson::to_string_view(x));
+    }
+}
+
+void tasks_set_ttl_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
+    if (!vm.contains("ttl")) {
+        throw std::invalid_argument("required parameter is missing: ttl");
+    }
+    auto new_ttl = vm["ttl"].as<uint32_t>();
+    std::unordered_map<sstring, sstring> params = {{ "ttl", fmt::to_string(new_ttl) }};
+
+    auto res = client.post("/task_manager/ttl", std::move(params));
+}
+
+void tasks_status_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
+    if (!vm.contains("id")) {
+        throw std::invalid_argument("required parameter is missing: id");
+    }
+    auto id = vm["id"].as<sstring>();
+
+    auto res = client.get(format("/task_manager/task_status/{}", id));
+
+    tasks_print_status(res);
+}
+
+void tasks_tree_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
+    std::vector<rjson::value> res;
+    if (vm.contains("id")) {
+        auto id = vm["id"].as<sstring>();
+
+        res.push_back(client.get(format("/task_manager/task_status_recursive/{}", id)));
+
+        tasks_print_trees(res);
+        return;
+    }
+
+    auto module_res = client.get("/task_manager/list_modules");
+    for (const auto& module : module_res.GetArray()) {
+        auto list_res = client.get(format("/task_manager/list_module_tasks/{}", module.GetString()));
+        for (const auto& stats : list_res.GetArray()) {
+            try {
+                res.push_back(client.get(format("/task_manager/task_status_recursive/{}", stats.GetObject()["task_id"].GetString())));
+            } catch (const api_request_failed& e) {
+                if (e.status == http::reply::status_type::bad_request) {
+                    // Task has already been unregistered.
+                    continue;
+                }
+                throw;
+            }
+        }
+    }
+    tasks_print_trees(res);
+}
+
+void tasks_ttl_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
+    auto res = client.get("/task_manager/ttl");
+    fmt::print("Current ttl: {}\n", res);
+}
+
+void tasks_wait_operation(scylla_rest_client& client, const bpo::variables_map& vm) {
+    if (!vm.count("id")) {
+        throw std::invalid_argument("required parameter is missing: id");
+    }
+    std::unordered_map<sstring, sstring> params;
+    auto id = vm["id"].as<sstring>();
+    bool quiet = vm.count("quiet");
+    if (vm.count("timeout")) {
+        params["timeout"] = fmt::format("{}", vm["timeout"].as<uint32_t>());
+    }
+
+    std::exception_ptr ex = nullptr;
+    try {
+        auto res = client.get(format("/task_manager/wait_task/{}", id), params);
+        if (quiet) {
+            throw operation_failed_with_status(res.GetObject()["state"] == "failed" ? 123 : 0);
```

Comment:
Why do we need to throw here if the task succeeded?
Can't we just `return` in that case?

BTW any `operation_failed_with_status` you throw from this block is caught below and re-thrown with status `125`, so the real status is lost.

We need tests for all the different cases to ensure they work.